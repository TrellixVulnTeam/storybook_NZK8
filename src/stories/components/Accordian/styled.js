// @flow

import styled from 'styled-components';
import downArrowIcon from 'assets/commonIcons/downArrow.svg';
import { colors } from 'utils/StyleGuide';

export const Wrapper = styled.div`
  box-shadow: ${p => (p.useDefaultStyle ? '2px 2px 5px rgba(0,0,0,.3)' : 'none')};
  border-radius: ${p => (p.useDefaultStyle ? '6px' : '0px')};
  border: ${p => (p.useDefaultStyle ? `solid 1px ${colors.paleGrey}` : 'none')};
`;

export const TitleBlock = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  cursor: pointer;
`;

export const Body = styled.div`
  background-color: ${colors.alabaster};
  height: ${p => (p.open.toString() === 'true' ? 'auto' : '0')};
  overflow: ${p => (p.open.toString() === 'true' ? 'visible' : 'hidden')};
`;

export const DownArrow = styled(downArrowIcon)`
  height: 32px;
  width: 32px;
  margin-right: 12px;
  fill: ${colors.silver};
  transform: rotate(${p => (p.up === 'true' ? '180deg' : '0deg')});
`;
