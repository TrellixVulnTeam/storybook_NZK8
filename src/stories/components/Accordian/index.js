// @flow

import React, { PureComponent } from 'react';
import { Wrapper, TitleBlock, Body, DownArrow } from './styled';

type Props = {
  header: React$Node,
  body: React$Node,
  alwaysOpen?: boolean,
  useDefaultStyle?: boolean,
  onAccordionToggle?: Function,
  defaultOpen?: boolean,
  className?: string,
};

type State = {
  expanded: boolean,
};

class Accordian extends PureComponent<Props, State> {
  // eslint-disable-next-line react/state-in-constructor
  state = {
    expanded: !!this.props.alwaysOpen || !!this.props.defaultOpen,
  };

  // eslint-disable-next-line react/static-property-placement
  static defaultProps = {
    useDefaultStyle: true,
  };

  handleOpenClose = () => {
    if (this.props.alwaysOpen) {
      return;
    }

    if (this.props.onAccordionToggle) {
      this.props.onAccordionToggle(!this.state.expanded);
    }
    // eslint-disable-next-line react/no-access-state-in-setstate
    this.setState({ expanded: !this.state.expanded });
  };

  /**
   * Render an accordian
   * @return {React$Node}
   */
  render() {
    return (
      <Wrapper useDefaultStyle={!!this.props.useDefaultStyle} className={this.props.className}>
        <TitleBlock onClick={this.handleOpenClose}>
          {this.props.header}
          {!this.props.alwaysOpen && this.props.useDefaultStyle && (
            <DownArrow up={this.state.expanded.toString()} />
          )}
        </TitleBlock>
        <Body open={!!this.state.expanded}>{this.props.body}</Body>
      </Wrapper>
    );
  }
}

export default Accordian;
