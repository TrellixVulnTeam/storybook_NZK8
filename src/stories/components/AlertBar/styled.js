// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import closeIcon from 'assets/commonIcons/closeIcon.svg';

/**
 * Determine the alert bar's accent color based on the "theme" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'none':
      return;
    case 'gray':
      return colors.silver;
    case 'green':
      return colors.mint;
    case 'blue':
      return colors.reallyLightBlue;
    case 'yellow':
      return colors.lightTan;
    case 'red':
      return colors.peach;
    default:
      return colors.peach;
  }
}

/**
 * Determine the alert bar's font color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getFontColor(theme) {
  switch (theme) {
    case 'none':
      return;
    case 'gray':
      return colors.white;
    case 'blue':
      return colors.dodgerBlue;
    case 'yellow':
      return colors.burntYellow;
    case 'green':
      return colors.jungleGreen;
    case 'red':
      return colors.strawberry;
    default:
      return colors.coral;
  }
}
export const CloseIcon = styled(closeIcon)`
  height: 12px;
  width: 12px;
  cursor: pointer;
  position: absolute;
  right: 20px;
`;

export const Wrapper = styled.div`
  display: ${p => (p.theme === 'none' ? 'none' : 'flex')};
  flex: 1;
  padding: 5px;
  background-color: ${p => getAccentColor(p.theme)};
  color: ${p => getFontColor(p.theme)};
  font-size: 14px;
  justify-content: center;
  align-items: center;
  padding-right: ${p => (p.dismissable ? '50px' : '0px')};
  a {
    color: ${p => getFontColor(p.theme)};
  }
  + ${CloseIcon} {
    fill: ${p => getFontColor(p.theme)};
  }
`;

export const OuterWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;
