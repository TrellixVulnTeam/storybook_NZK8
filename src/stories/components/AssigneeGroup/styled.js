// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

export const AssigneeWrapper = styled.div`
  display: flex;
  align-items: center;

  ul {
    display: inline-flex;
    padding: 0;
    margin: 0;
  }

  /* NumberBadge */
  span {
    position: relative;
    left: -24px;
    top: 10px;
    z-index: 10;

    @media (max-width: 1445px) and (min-width: 1200px) {
      display: none;
    }

    @media (max-width: 1185px) {
      display: none;
    }
  }
`;

export const Assignee = styled.li`
  width: ${p => (p.size === 'sm' ? '24px' : '36px')};
  height: ${p => (p.size === 'sm' ? '24px' : '36px')};
  line-height: ${p => (p.size === 'sm' ? '22px' : '36px')};
  text-align: center;
  font-size: ${p => p.size === 'sm' && '10px'};
  display: inline-block;
  border: 1px solid ${colors.white};
  background-image: url(${props => props.avatar});
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
  background-origin: border-box;
  background-color: ${props => props.assigneeColor};
  color: ${colors.white} !important;
  border-radius: 50%;

  &:first-of-type {
    z-index: ${props => props.offset && props.offset};
  }

  &:not(:first-of-type) {
    transform: translateX(${props => props.offset * props.index * -1}px);
    z-index: ${props => props.offset && (props.index - props.offset) * -1};

    @media (max-width: 1350px) and (min-width: 1200px) {
      display: none;
    }

    @media (max-width: 1090px) {
      display: none;
    }
  }

  svg {
    position: relative;
    top: -1px;
    left: -1px;
    width: 36px;
    height: 36px;
  }
`;
