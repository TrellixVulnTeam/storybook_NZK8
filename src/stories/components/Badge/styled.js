// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

/**
 * Determine the badge's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'none':
      return;
    case 'gray':
      return colors.paleGrey;
    case 'green':
      return colors.shamrockGreen;
    case 'blue':
      return colors.iceBlue;
    case 'yellow':
      return colors.lightTan;
    default:
      return colors.peach;
  }
}

/**
 * Determine the badge's font color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getFontColor(theme) {
  switch (theme) {
    case 'none':
      return;
    case 'gray':
      return colors.white;
    case 'blue':
      return colors.dodgerBlue;
    case 'yellow':
      return colors.burntYellow;
    case 'green':
      return colors.white;
    default:
      return colors.coral;
  }
}

export const Wrapper = styled.div`
  display: ${p => (p.theme === 'none' ? 'none' : 'inline-block')};
  padding: 0 8px;
  height: 24px;
  border-radius: 4px;
  background-color: ${p => getAccentColor(p.theme)};
  color: ${p => getFontColor(p.theme)};
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.3px;
  text-align: center;
`;
