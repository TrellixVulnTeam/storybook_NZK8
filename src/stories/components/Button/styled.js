// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { lighten } from 'polished';
import { colors } from 'utils/StyleGuide';
import { getGradientColors } from 'utils/StyleHelpers';

/**
 * Determine the button's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'blue':
      return colors.brightBlue;
    case 'green':
      return colors.shamrockGreen;
    case 'dark-green':
      return colors.jungleGreen;
    case 'white':
      return colors.alabaster;
    case 'red':
      return colors.strawberry;
    case 'silver':
      return colors.silver;
    case 'cancel':
    case 'transparent':
    case 'filter':
    case 'filter-active':
    case 'blue-bordered':
    case 'steel':
    default:
      return colors.transparent;
  }
}

/**
 * Determine the button's font color based on the "theme" prop
 * @param {string} theme
 * @return {string}
 */
function getFontColor(theme) {
  switch (theme) {
    case 'transparent':
    case 'blue-bordered':
      return colors.brightBlue;
    case 'steel':
      return colors.steel;
    case 'red':
    case 'blue':
    case 'green':
    case 'blue-gradient':
    case 'red-gradient':
    case 'dark-green':
    case 'white-outline':
      return colors.white;
    case 'cancel':
    case 'white':
    case 'filter':
    case 'filter-active':
    default:
      return colors.slateGrey;
  }
}

/**
 * Determine the button's font color
 * on hover based on the "theme" prop
 * @param {string} theme
 * @return {string}
 */
function getHoverFontColor(theme) {
  switch (theme) {
    case 'red':
    case 'blue':
    case 'green':
    case 'dark-green':
    case 'blue-gradient':
    case 'red-gradient':
    case 'white-outline':
      return colors.white;
    case 'white':
    case 'cancel':
    case 'transparent':
    case 'filter':
    case 'filter-active':
    case 'blue-bordered':
    case 'steel':
    default:
      return colors.brightBlue;
  }
}

/**
 * Determine the button's border color based on the "theme" prop
 * @param {string} theme
 * @return {string}
 */
function getBorderColor(theme) {
  switch (theme) {
    case 'white':
      return '#ccc';
    case 'filter':
      return colors.silver;
    case 'filter-active':
    case 'blue-bordered':
      return colors.brightBlue;
    case 'white-outline':
      return colors.white;
    default:
      return '';
  }
}

/**
 * Determine the button's size based on the "size" prop
 * @param {string} size
 * @return {string}
 */
function getSize(size) {
  switch (size) {
    case 'xs':
      return { width: 36, height: 36, font: 12, vpad: 0, hpad: 0 };
    case 'sm':
      return { width: 60, height: 36, font: 12, vpad: 4, hpad: 8 };
    case 'lg':
      return { width: 70, height: 48, font: 16, vpad: 6, hpad: 20 };
    case 'xlg':
      return { width: 80, height: 42, font: 14, vpad: 8, hpad: 15 };
    case 'xl':
      return { width: 100, height: 60, font: 20, vpad: 8, hpad: 24 };
    case 'xxl':
      return { width: 150, height: 42, font: 14, vpad: 10, hpad: 42 };
    case 'xxxl':
      return { width: 360, height: 50, font: 16, vpad: 0, hpad: 0 };
    default:
      return { width: 100, height: 36, font: 14, vpad: 6, hpad: 16 };
  }
}

/**
 * Determine the button's box-shadow
 *  on hover based on the "theme" prop
 * @param {string} theme
 * @return {string} box-shadow property value
 */
function getBoxShadow(theme) {
  switch (theme) {
    case 'filter':
    case 'filter-active':
    case 'transparent':
      return '0 2px 10px 0 rgba(40, 44, 49, 0.2)';
    default:
      return 'none';
  }
}

/**
 * Determine the button's icon
 * color based on the "theme" prop
 * @param {string} theme
 * @return {string} fill property value
 */
function getFillColor(theme) {
  switch (theme) {
    case 'cancel':
    case 'blue-bordered':
      return colors.brightBlue;
    case 'red':
    case 'green':
    case 'dark-green':
    case 'blue':
    case 'red-gradient':
    case 'blue-gradient':
      return colors.white;
    case 'white':
    case 'transparent':
    case 'filter':
    case 'filter-active':
    case 'steel':
    default:
      return colors.steel;
  }
}

/**
 * Determine the button's icon
 * color based on the "theme" prop
 * @param {string} theme
 * @return {string} fill property value
 */
function getHoverFillColor(theme) {
  switch (theme) {
    case 'red':
    case 'green':
    case 'dark-green':
    case 'blue':
    case 'red-gradient':
    case 'blue-gradient':
      return colors.white;
    case 'white':
    case 'transparent':
    case 'filter':
    case 'filter-active':
    case 'cancel':
    case 'blue-bordered':
    case 'steel':
    default:
      return colors.brightBlue;
  }
}

export const Wrapper = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  position: relative;
  margin: 0 6px;
  color: ${p => getFontColor(p.theme)};
  font-size: ${p => getSize(p.size).font}px;
  font-weight: ${p => (p.thin ? 400 : 600)};
  height: ${p => getSize(p.size).height}px;
  padding: ${p => getSize(p.size).vpad}px ${p => getSize(p.size).hpad}px;
  border-radius: ${p => (p.pill ? 30 : 6)}px;
  text-align: center;
  border: ${p =>
    getBorderColor(p.theme)
      ? `1px solid ${getBorderColor(p.theme)}`
      : `1px solid ${colors.transparent}`};
  background-clip: padding-box;
  background-color: ${({ theme, disabledBackground, disabled }) =>
    disabled && disabledBackground ? getAccentColor(disabledBackground) : getAccentColor(theme)};
  background-image: ${p =>
    getGradientColors(p.theme).length === 2
      ? `linear-gradient(251deg, ${getGradientColors(p.theme)[0]}, ${
          getGradientColors(p.theme)[1]
        })`
      : 'none'};
  opacity: ${p => (!p.loading && p.disabled ? 0.5 : 1)};
  letter-spacing: -0.34px;
  min-width: ${p => getSize(p.size).width}px;
  transition: all 150ms ease;

  svg {
    fill: ${p => getFillColor(p.theme)};
    margin-right: 5px;
  }

  &:hover {
    cursor: ${p => (p.disabled ? 'default' : 'pointer')};
    background-color: ${p =>
      p.theme === 'cancel' || p.theme === 'white-outline'
        ? 'rgba(255,255,255,0.12)'
        : lighten(0.15, getAccentColor(p.theme))};
    box-shadow: ${p => getBoxShadow(p.theme)};
    color: ${p => getHoverFontColor(p.theme)};
    background-image: ${p =>
      getGradientColors(p.theme).length === 2
        ? `linear-gradient(251deg, ${lighten(0.15, String(getGradientColors(p.theme)[0]))},
      ${lighten(0.15, String(getGradientColors(p.theme)[1]))})`
        : 'none'};
    border: ${p => p.theme === 'filter' && '1px solid transparent'};

    svg {
      fill: ${p => getHoverFillColor(p.theme)};
    }
  }

  &:focus {
    text-decoration: none;
    outline: none;
  }
`;
