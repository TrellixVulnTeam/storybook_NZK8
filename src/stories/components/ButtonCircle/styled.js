// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';

/**
 * Determine the button's accent color based on the "color" prop
 * @param {string} theme
 * @return {string}
 */
function getAccentColor(theme) {
  switch (theme) {
    case 'blue':
      return colors.brightBlue;
    case 'red':
      return colors.strawberry;
    default:
      return '';
  }
}

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.button`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 34px;
  height: 34px;
  line-height: 34px;
  background: ${p => getAccentColor(p.theme)};

  border-radius: 50%;
  color: ${colors.white};
  cursor: pointer;

  span {
    width: 100%;
    text-align: center;
  }

  &:hover {
    box-shadow: 0 2px 10px 0 rgba(40, 44, 49, 0.2);
  }
`;
