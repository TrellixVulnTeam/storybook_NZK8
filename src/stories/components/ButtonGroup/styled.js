// @flow

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import scrumIcon from 'assets/commonIcons/scrumIcon.svg';
import calendarIcon from './img/calendarIcon.svg';
import listIcon from './img/listIcon.svg';

/**
 * Determine the button group's size based on the "size" prop
 * @param {string} size
 * @return {string}
 */
function getSize(size) {
  switch (size) {
    case 'xs':
      return { width: 36, height: 30, font: 10, vpad: 0, hpad: 4 };
    case 'sm':
      return { width: 36, height: 36, font: 11, vpad: 0, hpad: 8 };
    case 'lg':
      return { width: 60, height: 48, font: 20, vpad: 0, hpad: 16 };
    default:
      return { width: 80, height: 36, font: 12, vpad: 0, hpad: 12 };
  }
}

export const Wrapper = styled.div`
  margin: 0;
  border-radius: 6px;
  display: ${p => (p.flex ? 'flex' : 'inline-flex')};
  height: ${p => getSize(p.size).height}px;

  [type='radio'] {
    display: none;
  }

  .radio-option {
    min-width: ${p => getSize(p.size).width}px;
    padding: ${p => getSize(p.size).vpad}px ${p => getSize(p.size).hpad}px;
    text-align: center;
    cursor: pointer;
    border: 1px solid #ccc;
    transition: 100ms;
    flex-grow: ${p => p.flex && 1};

    span {
      line-height: ${p => getSize(p.size).height}px;
      font-size: ${p => getSize(p.size).font}px;
    }

    &.disabled {
      background: #ddd;

      span {
        color: #aaa;
      }
    }
    svg {
      fill: ${colors.silver};
    }
  }

  .radio-option:hover:not(.disabled):not(.selected) {
    background: ${colors.white};
    transition: 100ms;
    .option-label {
      color: ${colors.brightBlue};
    }

    svg {
      fill: ${colors.brightBlue};
    }
  }

  .radio-option.selected {
    transition: 100ms;
    background: ${colors.iceBlue};

    &.disabled {
      background-image: linear-gradient(251deg, #888, #777);
    }

    span {
      color: ${colors.brightBlue};
    }

    svg {
      fill: ${colors.brightBlue};
    }
  }

  .tooltip-option {
    flex-grow: ${p => p.flex && 1};

    &:first-child {
      .radio-option {
        border-radius: 6px 0 0 6px;
      }
    }

    &:not(:first-child):not(:last-child) {
      .radio-option {
        border-left: none;
      }
    }

    &:last-child {
      .radio-option {
        border-radius: 0 6px 6px 0;
        border-left: none;
      }
    }
  }

  .option-label {
    color: ${colors.silver};
    font-size: 12px;
  }
`;

export const CalendarIcon = styled(calendarIcon)`
  width: 16px;
  height: 16px;
  margin-top: 10px;
`;

export const ListIcon = styled(listIcon)`
  width: 16px;
  height: 16px;
  margin-top: 10px;
`;

export const ScrumIcon = styled(scrumIcon)`
  width: 16px;
  height: 16px;
  margin-top: 10px;
`;
