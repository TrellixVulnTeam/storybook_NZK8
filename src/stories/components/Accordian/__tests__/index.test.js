import React from 'react';
import { mount } from 'enzyme';
import Accordian from 'components/Accordian/index';
import { TitleBlock } from 'components/Accordian/styled';

describe('<Accordian />', () => {
  const Header = () => <p>Header</p>;
  const Body = () => <p>Body</p>;
  const onAccordionToggleMock = jest.fn();
  const mandatoryProps = {
    header: Header,
    body: Body,
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with only mandatory props given', () => {
    const sut = mount(<Accordian {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with all props given', () => {
    const props = {
      ...mandatoryProps,
      alwaysOpen: true,
      useDefaultStyle: false,
      onAccordionToggle: onAccordionToggleMock,
      constdefaultOpen: true,
      className: '',
    };
    const sut = mount(<Accordian {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should correctly toggle without custom handler', () => {
    const sut = mount(<Accordian {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();

    sut.find(TitleBlock).simulate('click');

    expect(sut).toMatchSnapshot();
  });

  it('should call custom handler on toggle', () => {
    const props = {
      ...mandatoryProps,
      onAccordionToggle: onAccordionToggleMock,
    };
    const sut = mount(<Accordian {...props} />);

    expect(sut).toMatchSnapshot();

    expect(onAccordionToggleMock).toHaveBeenCalledTimes(0);

    sut.find(TitleBlock).simulate('click');

    expect(sut).toMatchSnapshot();

    expect(onAccordionToggleMock).toHaveBeenCalledTimes(1);
  });

  it('should not toggle if always open', () => {
    const props = {
      ...mandatoryProps,
      alwaysOpen: true,
    };
    const sut = mount(<Accordian {...props} />);

    expect(sut).toMatchSnapshot();

    sut.find(TitleBlock).simulate('click');

    expect(sut).toMatchSnapshot();
  });
});
