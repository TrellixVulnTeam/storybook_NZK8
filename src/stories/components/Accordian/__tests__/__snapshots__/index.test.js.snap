// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordian /> should call custom handler on toggle 1`] = `
<Accordian
  body={[Function]}
  header={[Function]}
  onAccordionToggle={[MockFunction]}
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <styled__DownArrow
            up="false"
          >
            <test-file-stub
              className="styled__DownArrow-exutvq-3 cOBOxj"
            />
          </styled__DownArrow>
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={false}
      >
        <div
          className="styled__Body-exutvq-2 kSOKag"
          open={false}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should call custom handler on toggle 2`] = `
<Accordian
  body={[Function]}
  header={[Function]}
  onAccordionToggle={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <styled__DownArrow
            up="true"
          >
            <test-file-stub
              className="styled__DownArrow-exutvq-3 Gkmrw"
            />
          </styled__DownArrow>
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={true}
      >
        <div
          className="styled__Body-exutvq-2 gyjJto"
          open={true}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should correctly toggle without custom handler 1`] = `
<Accordian
  body={[Function]}
  header={[Function]}
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <styled__DownArrow
            up="false"
          >
            <test-file-stub
              className="styled__DownArrow-exutvq-3 cOBOxj"
            />
          </styled__DownArrow>
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={false}
      >
        <div
          className="styled__Body-exutvq-2 kSOKag"
          open={false}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should correctly toggle without custom handler 2`] = `
<Accordian
  body={[Function]}
  header={[Function]}
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <styled__DownArrow
            up="true"
          >
            <test-file-stub
              className="styled__DownArrow-exutvq-3 Gkmrw"
            />
          </styled__DownArrow>
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={true}
      >
        <div
          className="styled__Body-exutvq-2 gyjJto"
          open={true}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should not toggle if always open 1`] = `
<Accordian
  alwaysOpen={true}
  body={[Function]}
  header={[Function]}
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <Component />
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={true}
      >
        <div
          className="styled__Body-exutvq-2 gyjJto"
          open={true}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should not toggle if always open 2`] = `
<Accordian
  alwaysOpen={true}
  body={[Function]}
  header={[Function]}
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <Component />
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={true}
      >
        <div
          className="styled__Body-exutvq-2 gyjJto"
          open={true}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should render correctly with all props given 1`] = `
<Accordian
  alwaysOpen={true}
  body={[Function]}
  className=""
  constdefaultOpen={true}
  header={[Function]}
  onAccordionToggle={[MockFunction]}
  useDefaultStyle={false}
>
  <styled__Wrapper
    className=""
    useDefaultStyle={false}
  >
    <div
      className="styled__Wrapper-exutvq-0 cCximc"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <Component />
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={true}
      >
        <div
          className="styled__Body-exutvq-2 gyjJto"
          open={true}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;

exports[`<Accordian /> should render correctly with only mandatory props given 1`] = `
<Accordian
  body={[Function]}
  header={[Function]}
  useDefaultStyle={true}
>
  <styled__Wrapper
    useDefaultStyle={true}
  >
    <div
      className="styled__Wrapper-exutvq-0 hRanGO"
    >
      <styled__TitleBlock
        onClick={[Function]}
      >
        <div
          className="styled__TitleBlock-exutvq-1 khWXmY"
          onClick={[Function]}
        >
          <styled__DownArrow
            up="false"
          >
            <test-file-stub
              className="styled__DownArrow-exutvq-3 cOBOxj"
            />
          </styled__DownArrow>
        </div>
      </styled__TitleBlock>
      <styled__Body
        open={false}
      >
        <div
          className="styled__Body-exutvq-2 kSOKag"
          open={false}
        >
          <Component />
        </div>
      </styled__Body>
    </div>
  </styled__Wrapper>
</Accordian>
`;
